### PR TITLE
Refactor to prepare for release

### PR DESCRIPTION
**Pull request by robertdp**
This is a long-overdue clean up of parts of the routing logic and underlying implementation.

@jonasbuntinx Care to give this a look-over?

The downstream changes are pretty mild: https://github.com/robertdp/purescript-web-router-example/pull/2/files#diff-eb73bc89dc00e8262fee8eaa74c28d14d54c2d71c3275faddd28870baa9370ea

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,30 +1,9 @@
 # purescript-web-router
 
-A basic pushstate router for React with support for asynchronous routing logic, built using [react-basic-hooks](https://github.com/spicydonuts/purescript-react-basic-hooks). I recommend [routing-duplex](https://github.com/natefaubion/purescript-routing-duplex) for easy parsing and printing.
+A basic pushstate router for React with support for asynchronous routing logic. I recommend [routing-duplex](https://github.com/natefaubion/purescript-routing-duplex) for easy parsing and printing.
```
Comment (paluh) [inline README.md]:
```suggestion
A basic pushstate router with support for asynchronous routing logic. I recommend [routing-duplex](https://github.com/natefaubion/purescript-routing-duplex) for easy parsing and printing.
```

File: packages.dhall [hunk 2]
```diff
@@ -113,14 +96,10 @@ let additions =
...
       }
-  }
 -------------------------------
 -}
 let upstream =
-      https://github.com/purescript/package-sets/releases/download/psc-0.14.0-20210311/packages.dhall sha256:3da8be2b7b4a0e7de6186591167b363023695accffb98a8639e9e7d06e2070d6
-
-let overrides = {=}
-
-let additions = {=}
```
Comment (paluh) [inline packages.dhall:124]: It would be nice to drop these `spago` comments all together because they clutter the actual diffs :-P

**Review by paluh — COMMENTED** [review]
Hi Robert!

Sorry for jumping ahead and commenting your PR but I want to say that I really like these changes. From a user perspective I can say that:

* `RouterM` and `Routing` is much cleaner naming convention than `Transition` and `Transitioning`
* The same goes for `RouterState` vs `RouterEvent`
* As a consequence the surface API of `mkRouter` is now beautiful and easy to understand :-)

Thanks a lot for this update!

**Comment by jonasbuntinx** [comment]
LGTM 👍

File: README.md (hunk 1 above)
Comment (robertdp) [reply README.md]: The readme looks... a little different now 🙂

File: packages.dhall (hunk 2 above)
Comment (robertdp) [reply packages.dhall:124]: That's a fair point. I've stripped them out.

File: README.md (hunk 1 above)
Comment (paluh) [reply README.md]: README looks great :-)

Do you think that making a README part of a "test suite" (so we verify whether it still compiles) is feasible and desired?

I mean I use this strategy across my libs just by using something like: 
https://github.com/paluh/purescript-homogeneous/blob/master/package.json#L10 (`litps` is really a tool written by [@Thimoteus](https://github.com/Thimoteus/literate-purescript/issues/2)).

Comment (robertdp) [reply README.md]: If PS supported literate code I'd consider it, but I don't think it's worth the extra effort. The example project using React has also been updated and it compiles, so most of the code in the updated readme was tested there first.

Some unit tests would be good, but can come a bit later.

Comment (paluh) [reply README.md]: This `litps` provides support for "literate PS". Maybe I will play around that idea in my... spare time ;-)

**Pull request merged**